### PR TITLE
Prepare data and quality check adapatation to data class

### PR DESCRIPTION
**Pull request by camillebrianceau**
- Adaptation of `CapsDataset` to `CapsDatasetConfig`
- Suppression of `PrepareDataConfig`
- `return_dataset` function back in `data.py` file
- adapt `cli.py` files of `prepare-data`, `prepare-data-from-bids` and `quality-check` to the new organization

NEED TO REBASE  !

### CODE REVIEW

File: clinicadl/caps_dataset/data_config.py [hunk 1]
```diff
@@ -35,15 +26,16 @@ class DataConfig(BaseModel):  # TODO : put in data module
...
-    label_code: Dict[str, int] = {}
+    label_code: Union[str, dict[str, int], None] = {}
     multi_cohort: bool = False
     mask_path: Optional[Path] = None
     preprocessing_json: Optional[Path] = None
     data_tsv: Optional[Path] = None
     n_subjects: int = 300
     # pydantic config
-    model_config = ConfigDict(validate_assignment=True)
+    model_config = ConfigDict(validate_assignment=True, arbitrary_types_allowed=True)
```
Comment (thibaultdvx) [inline clinicadl/caps_dataset/data_config.py:38]: What does `arbitrary_types_allowed` do? Does it disable all Pydantic's type checks?

File: clinicadl/quality_check/t1_linear/quality_check.py [hunk 2]
```diff
@@ -126,13 +134,16 @@ def quality_check(
 
     with torch.no_grad():
         # Transform caps_dir in dict
-        caps_dict = CapsDataset.create_caps_dict(caps_dir, multi_cohort=False)
 
+        caps_dict = config.data.caps_dict
+        print(caps_dict)
```
Comment (thibaultdvx) [inline clinicadl/quality_check/t1_linear/quality_check.py]:
```suggestion
```

**Review by thibaultdvx — COMMENTED** [review]
Just spotted one wild `print`.  LGTM otherwise!

File: clinicadl/caps_dataset/data_config.py (hunk 1 above)
Comment (camillebrianceau) [reply clinicadl/caps_dataset/data_config.py:38]: I think it is just to accept other types like `pd.DataFrame` and not only basic ones!

**Pull request merged**